dotCMS/src/integration-test/java/com/dotcms/rest/api/v1/secret/ServiceIntegrationResourceTest.java

```diff
@@ -0,0 +1,428 @@
+package com.dotcms.rest.api.v1.secret;
+
+import static com.dotcms.unittest.TestUtil.upperCaseRandom;
+import static org.mockito.Matchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import com.dotcms.IntegrationTestBase;
+import com.dotcms.datagen.SiteDataGen;
+import com.dotcms.repackage.org.apache.commons.httpclient.HttpStatus;
+import com.dotcms.rest.InitDataObject;
+import com.dotcms.rest.ResponseEntityView;
+import com.dotcms.rest.WebResource;
+import com.dotcms.rest.api.v1.secret.view.ServiceIntegrationDetailedView;
+import com.dotcms.rest.api.v1.secret.view.ServiceIntegrationSiteView;
+import com.dotcms.rest.api.v1.secret.view.ServiceIntegrationView;
+import com.dotcms.rest.api.v1.secret.view.SiteView;
+import com.dotcms.security.secret.Param;
+import com.dotcms.security.secret.Secret;
+import com.dotcms.security.secret.ServiceDescriptor;
+import com.dotcms.security.secret.Type;
+import com.dotcms.util.IntegrationTestInitService;
+import com.dotmarketing.beans.Host;
+import com.fasterxml.jackson.annotation.JsonAutoDetect.Visibility;
+import com.fasterxml.jackson.annotation.PropertyAccessor;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
+import com.google.common.collect.ImmutableMap;
+import com.liferay.portal.model.User;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Map.Entry;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.ws.rs.core.Response;
+import org.glassfish.jersey.media.multipart.ContentDisposition;
+import org.glassfish.jersey.media.multipart.FormDataBodyPart;
+import org.glassfish.jersey.media.multipart.FormDataMultiPart;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class ServiceIntegrationResourceTest extends IntegrationTestBase {
+
+    static final String ADMIN_DEFAULT_ID = "dotcms.org.1";
+    static final String ADMIN_DEFAULT_MAIL = "admin@dotcms.com";
+    static final String ADMIN_NAME = "User Admin";
+
+    private static ServiceIntegrationResource serviceIntegrationResource;
+
+    private static final ObjectMapper ymlMapper = new ObjectMapper(new YAMLFactory())
+            //.enable(SerializationFeature.INDENT_OUTPUT)
+            .setVisibility(PropertyAccessor.FIELD, Visibility.ANY)
+            .findAndRegisterModules()
+            ;
+
+    @BeforeClass
+    public static void prepare() throws Exception {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4623512)